src/heur.rs

```diff
@@ -0,0 +1,172 @@
+    pub const AFTER_LP_LOOP: Self = HeurTiming(0x004);
+    /// call heuristic after the processing of a node with solved LP was finished
+    pub const AFTER_LP_NODE: Self = HeurTiming(0x008);
+    /// call heuristic after the processing of a node without solved LP was finished
+    pub const AFTER_PSEUDO_NODE: Self = HeurTiming(0x010);
+    /// call heuristic after the processing of the last node in the current plunge was finished, and only if the LP was solved for this node
+    pub const AFTER_LP_PLUNGE: Self = HeurTiming(0x020);
+    /// call heuristic after the processing of the last node in the current plunge was finished, and only if the LP was not solved for this node
+    pub const AFTER_PSEUDO_PLUNGE: Self = HeurTiming(0x040);
+    /// call heuristic during pricing loop
+    pub const DURING_PRICING_LOOP: Self = HeurTiming(0x080);
+    /// call heuristic before presolving
+    pub const BEFORE_PRESOL: Self = HeurTiming(0x100);
+    /// call heuristic during presolving loop
+    pub const DURING_PRESOL_LOOP: Self = HeurTiming(0x200);
+    /// call heuristic after propagation which is performed before solving the LP
+    pub const AFTER_PROP_LOOP: Self = HeurTiming(0x400);
+}
+
+impl BitOr for HeurTiming {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self::Output {
+        HeurTiming(self.0 | rhs.0)
+    }
+}
+
+impl BitOrAssign for HeurTiming {
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.0 |= rhs.0;
+    }
+}
+
+impl From<HeurTiming> for u32 {
+    fn from(mask: HeurTiming) -> Self {
+        mask.0 as u32
+    }
+}
+
+impl From<u32> for HeurTiming {
+    fn from(mask: u32) -> Self {
+        HeurTiming(mask as u64)
+    }
+}
+
+impl From<HeurResult> for u32 {
+    fn from(val: HeurResult) -> Self {
+        match val {
+            HeurResult::FoundSol => ffi::SCIP_Result_SCIP_FOUNDSOL,
+            HeurResult::NoSolFound => ffi::SCIP_Result_SCIP_DIDNOTFIND,
+            HeurResult::DidNotRun => ffi::SCIP_Result_SCIP_DIDNOTRUN,
+            HeurResult::Delayed => ffi::SCIP_Result_SCIP_DELAYED,
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod tests {
+    use crate::Model;
+
+    use super::*;
+
+    struct NoSolutionFoundHeur;
+
+    impl Heur for NoSolutionFoundHeur {
+        fn execute(&mut self, _timing: HeurTiming, _node_inf: bool) -> HeurResult {
+            HeurResult::NoSolFound
+        }
+    }
+
+    #[test]
+    fn test_heur() {
+        let model = Model::new()
+            .hide_output()
+            .include_default_plugins()
+            .read_prob("data/test/simple.lp")
+            .unwrap();
+
+        let heur = NoSolutionFoundHeur;
+        model.include_heur(
+            "no_sol_found_heur",
+            "",
+            9999999,
+            'n',
+            1,
+            0,
+            -1,
+            HeurTiming::BEFORE_NODE,
+            false,
+            Box::new(heur),
+        ).solve();
+
+    }
+
+    struct LyingHeur;
+
+    impl Heur for LyingHeur {
```

Comment:
I would have preferred a full `Heuristic` trait name but Heur is fine

Reply:
I followed the PySCIPOpt naming :) but you're right clarity should win here, will change it!